org-code-javabuilder/theater/src/main/java/org/code/theater/Scene.java

```diff
@@ -0,0 +1,436 @@
+            endX,
+            endY,
+            this.strokeColor == null ? DEFAULT_COLOR : this.strokeColor,
+            this.strokeWidth));
+  }
+
+  /**
+   * Draw a regular polygon on the canvas.
+   *
+   * @param x the center X coordinate of the polygon
+   * @param y the center Y coordinate of the polygon
+   * @param sides the number of sides of the polygon
+   * @param radius the distance from the center to each point on the polygon
+   */
+  public final void drawRegularPolygon(int x, int y, int sides, int radius) {
+    this.actions.add(
+        new DrawPolygonAction(
+            x, y, sides, radius, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Draw as a shape by connecting the points provided.
+   *
+   * @param points an array of numbers representing the points. For instance, a triangle could be
+   *     represented as [x1, y1, x2, y2, x3, y3].
+   * @param close whether to close the shape. If this is set to true, the last point and the first
+   *     point will be connected by a line, and if a fill color is set, the shape will be filled
+   *     with that color.
+   */
+  public final void drawShape(int[] points, boolean close) {
+    this.actions.add(
+        new DrawShapeAction(points, close, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Draws an ellipse (an oval or a circle) on the canvas.
+   *
+   * @param x the left side of the ellipse
+   * @param y the top of the ellipse
+   * @param width the width of the ellipse
+   * @param height the height of the ellipse
+   */
+  public final void drawEllipse(int x, int y, int width, int height) {
+    this.actions.add(
+        new DrawEllipseAction(
+            x, y, width, height, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Draws a rectangle on the canvas.
+   *
+   * @param x the left side of the rectangle
+   * @param y the top of the rectangle
+   * @param width the width of the rectangle
+   * @param height the height of the rectangle
+   */
+  public final void drawRectangle(int x, int y, int width, int height) {
+    this.actions.add(
+        new DrawRectangleAction(
+            x, y, width, height, this.strokeColor, this.fillColor, this.strokeWidth));
+  }
+
+  /**
+   * Sets the thickness of lines drawn. Note that this only applies to drawing operations performed
+   * by this scene.
+   *
+   * @param width width in pixels of the line to draw. Zero means no line.
+   */
+  public final void setStrokeWidth(double width) {
+    this.strokeWidth = width;
+  }
+
+  /**
+   * Sets the fill color for all shapes drawn. Note that this only applies to drawing operations
+   * performed by this scene.
+   *
+   * @param color the color value to fill any shape.
+   */
+  public final void setFillColor(Color color) {
+    this.fillColor = color;
+  }
+
+  /**
+   * Sets the color of lines drawn. Note that this only applies to drawing operations performed by
+   * this actor.
```

Comment:
NIT: `actor` is still shown here. Perhaps "displayed by this scene" would be better? (same for the next several methods)